app/controllers/users/password_compromised_controller.rb

```diff
@@ -2,18 +2,53 @@
 
 module Users
   class PasswordCompromisedController < ApplicationController
+    include PasswordConcern
     before_action :confirm_two_factor_authenticated
     before_action :verify_feature_toggle_on
 
     def show
-      session.delete(:redirect_to_password_compromised)
-      @after_sign_in_path = after_sign_in_path_for(current_user)
+      session.delete(:redirect_to_change_password)
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user)
+      @forbidden_passwords = forbidden_passwords
       analytics.user_password_compromised_visited
     end
 
+    def update
+      @update_user_password_form = UpdateUserPasswordForm.new(current_user, user_session)
+
+      result = @update_user_password_form.submit(user_password_params)
+      updated_result = result.to_h.merge({ original_password_compromised: true })
+      analytics.password_changed(**updated_result)
+      if result.success?
+        handle_valid_password
+      else
+        handle_invalid_password
+      end
+    end
+
     def verify_feature_toggle_on
       redirect_to after_sign_in_path_for(current_user) unless
         FeatureManagement.check_password_enabled?
     end
+
+    def handle_valid_password
+      send_password_reset_risc_event
+      create_event_and_notify_user_about_password_change
+
+      bypass_sign_in current_user
```

Comment:
Can we copy over the code comment from `PasswordsController`? This code was very suspect otherwise.

https://github.com/18F/identity-idp/blob/25e66ea77a99ea4a6d87394dfd453c2f6e0a5509/app/controllers/users/passwords_controller.rb#L49-L50